REPORTING_ISSUES.md

```diff
@@ -0,0 +1,174 @@
+    MediaType/Media Type: *Stationery StationeryLightweight Midweight StationeryHeavyweight ExtraHeavy ColorTransparency Labels StationeryLetterhead Envelope StationeryPreprinted StationeryPrepunched Color Bond Recycled Rough Vellum
+    cupsPrintQuality/cupsPrintQuality: Draft *Normal
+    ColorModel/Output Mode: *Gray
+    Duplex/Duplex: *None DuplexNoTumble DuplexTumble
+    OutputBin/OutputBin: *FaceDown
+```
+
+Print a file to a printer with specific options (in case the printout from application is incorrect - print the same file and set the same options as you did in application, in case they are seen in `lpoptions`):
+
+```
+    $ lp -d HP_LaserJet_M1536dnf_MFP_42307C -o Duplex=DuplexNoTumble /etc/services
+```
+
+For more info check `man lp`, `man lpstat`, `man lpoptions`.
+
+2. check if the issue happens with different documents
+3. check if the issue happens when using different applications
+4. in case you use mDNS hostnames (hostnames with `.local`), check mDNS resolution by pinging such hostname - mDNS is used for temporary queues (printers which are seen by `lpstat -e`, but not by `lpstat -a`) or for permanent queues which have `.local` in its URI (check `lpstat -a`). However the hostname from URI is not resolvable as it is, you need to get usual printer's mDNS hostname from `hostname` entry in `avahi-browse -avrt` output.
+
+Rule of thumb:
+1. in case printing works from CUPS CLI tools or CUPS Web UI, but not via an application, file the issue to the application for the initial investigation.
+2. in case mDNS resolution doesn't work and you use driverless printing or mDNS hostnames in CUPS, file the issue to your mDNS resolution provider - it can be `nss-mdns` or `systemd-resolved` depending on your configuration.
+
+* turn on **CUPS debug logging**:
+
+```
+    $ cupsctl LogLevel=debug2
+```
+
+* **reproduce the issue**
+* **collect the logs** - the logs can be in `/var/log/cups/error_log` or in `journalctl`:
+
+For `error_log`:
+```
+    $ sudo cp /var/log/cups/error_log ~/error_log.txt
+    $ sudo chmod 666 ~/error_log.txt
+```
+
+For `journalctl`:
+```
+    $ journalctl -u cups --since=today > log.txt
+```
+
+and attach the file to the GitHub issue.
+
+* provide output of **lpstat -e** and **lpstat -t**
+* provide **PPD file** from `/etc/cups/ppd` if exists for the printer
+* provide **the file you are trying to print**, if the issue happens with a specific file
+* provide **the d file from /var/spool/cups** - this is the file CUPS actually gets from the application
+* tell us the name of application where you experience the problem
+* if needed, turn off the debug logging by:
+
+```
+    $ cupsctl LogLevel=warn
+```
+
+### INFORMATION REQUESTED FOR USB DEVICES
+
+* attach **output of `lsusb -v`** in a file as attachment
+* in case of **communication issues with USB device**, it is helpful to **capture USB communication** - you just has to know bus number where your device is connected to with `tcpdump`:
```

Comment:
Typo:
I think
"you just has to know"
should be
"you just have to know"

I would like to recommend
to leave out words like "just", "simple", "easy", ...
because when printing doesn't work the user is often
already desperate and then such words could implicitly
treat him as if he was an idiot because such words
implicitly tell things were "just" "simple" "easy"
but in fact printing is not "just" "simple" "easy"
as we all know too well from hundreds or thousands
of user issues we had handled in the past.

Reply:
Ok, I'll take that in mind - that `just` was meant like `one little thing is needed to make it work`, but that was just me trying to make it more human instead of having it as a plain documentation.

Typo fixed.

Reply:
I know that it is meant this way
but it could be percieved totally different
in particular by a desperate and/or frustrated user.

In general in technical documentation one should
keep it sound neutral and avoid a personal tone.
So neither a military parade ground commanding tone
nor as if you talk to someone personally.